corehq/form_processor/models.py

```diff
@@ -558,7 +558,7 @@ def get_xml(self):
     def xml_md5(self):
         return self.get_attachment_meta('form.xml').content_md5()
 
-    def archive(self, user_id=None):
+    def archive(self, user_id=None, skip_rebuild_cases=False):
```

Comment:
what do you think of renaming this to `bulk_operation` since it now also encompasses ledgers. That also gives more information on why you want to skip the rebuilds.

Reply:
this was named back to `rebuild_cases`.
How about `reprocess_cases_and_ledgers` or `update_cases_and_ledgers`?

Reply:
Another approach here would be to drop the signal entirely and call `rebuild_form_cases` and `process_form_archived` directly in this archive method.  That'd make it a bit more explicit what's being skipped, and would head off future issues for other prospective signal receivers like Manish mentioned, as the author would need to decide where in the `archive` method the call goes.

Though maybe that's just my anti-signal bias coming through.

Reply:
That was also my suggestion originally and I do still prefer that approach.

Reply:
Updated.
keeping signals as it is but skipped calling them completely.